Sources/Swift/Tools/ViewCapture/SentryViewPhotographer.swift

```diff
@@ -16,25 +16,31 @@ class SentryViewPhotographer: NSObject, SentryViewScreenshotProvider {
 
     init(
         renderer: SentryViewRenderer,
-        redactOptions: SentryRedactOptions
+        redactOptions: SentryRedactOptions,
+        enableExperimentalMasking: Bool
     ) {
         self.renderer = renderer
-        self.maskRenderer = SentryDefaultMaskRenderer()
+        self.maskRenderer = enableExperimentalMasking ? SentryExperimentalMaskRenderer() : SentryDefaultMaskRenderer()
         redactBuilder = UIRedactBuilder(options: redactOptions)
         super.init()
     }
 
     func image(view: UIView, onComplete: @escaping ScreenshotCallback) {
         let viewSize = view.bounds.size
         let redact = redactBuilder.redactRegionsFor(view: view)
+        // The render method is synchronous and must be called on the main thread.
+        // This is because the render method accesses the view hierarchy which is managed from the main thread.
         let renderedScreenshot = renderer.render(view: view)
 
         dispatchQueue.dispatchAsync { [maskRenderer] in
+            // The mask renderer does not need to be on the main thread.
+            // Moving it to a background thread to avoid blocking the main thread, therefore reducing the performance
+            // impact/lag of the user interface.
             let maskedScreenshot = maskRenderer.maskScreenshot(screenshot: renderedScreenshot, size: viewSize, masking: redact)
             onComplete(maskedScreenshot)
         }
     }
-    
+
     func image(view: UIView) -> UIImage {
```

Comment:
`m`: Can't we also move masking the screenshot here to a bg thread? Currenlty this is only called from capturing a screenshot, so maybe adding a comment explaining why we don't move it to a bg thread here would be helpful.

Reply:
Most likely yes, but this PR focuses on session replay and refactoring this method to be async too (= dropping it in favor of the other one), would require refactoring in the screenshot integration which is out of scope for this PR